### PR TITLE
chore: setup jest for unit test hooks and component from joi

### PR DESCRIPTION
**Pull request by urmauur**
## Describe Your Changes

- WIP unit test component 
Screenshot 2024-09-03 at 20 02 43

Screenshot 2024-09-03 at 20 03 03

## Fixes Issues

- #3507 
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: joi/tsconfig.json [hunk 1]
```diff
@@ -11,6 +12,6 @@
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "./jest.setup.js"],
```
Comment (louis-jan) [inline joi/tsconfig.json:15]: Should this JS file not be included in the tsconfig?

Comment (urmauur) [reply joi/tsconfig.json]: my bad, removed on next commit

**Review by louis-jan — APPROVED** [review]
LGTM

**Comment by louis-jan** [comment]
Please proceed, @urmauur. I will run the code coverage report, including the Joi project.

**Pull request merged**